### PR TITLE
Mooreds/1951 upgrade guide

### PR DESCRIPTION
**Pull request by mooreds**
Small changes, primarily based on this PR: https://github.com/FusionAuth/fusionauth-site/pull/2166

### CODE REVIEW

**Review by spwitt — CHANGES_REQUESTED** [review]
I made a lot of recommendations. Most of them are aimed at shortening or simplifying content. I'll let you decide which recommendations to take. Please let me know if you have questions about any of the recommendations.

**Comment by mooreds** [comment]
@spwitt do you want to re-review after I've made changes based on your recommendations?

File: site/docs/v1/tech/admin-guide/upgrade.adoc [hunk 1]
```diff
@@ -4,44 +4,371 @@ title: Upgrade FusionAuth
...
+** <<Docker>>
+** <<Homebrew>>
+** <<ZIP Packages>>
+** <<Linux Packages>>
+** <<FastPath>>
 
-If you used the FastPath or other installation methods initially, you may prefer to use that same mode to download the latest version of FusionAuth. If you would like to continue by manually downloading the necessary packages, read on.
+== When to Upgrade
+
+=== Reasons To Upgrade
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/upgrade.adoc:42]: All at once, later.

File: site/docs/v1/tech/admin-guide/upgrade.adoc [hunk 2]
```diff
@@ -4,44 +4,373 @@ title: Upgrade FusionAuth
...
+[WARNING]
+====
+Maintenance Mode should never be used in production. You don't want your end users to be prompted to update a database, which is what Maintenance Mode does.
+
+You should always back up your database before upgrading regardless of the upgrade method you choose.
+====
+
+=== Easiest Route
+
+The easiest upgrade route is to simply stop all your services, upgrade each FusionAuth node, and then bring it all back online. In this case, FusionAuth will be performing the database upgrade, if needed, while the servers are stopped.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/upgrade.adoc:146]: Hmm. It actually doesn't do the upgrade when the services are stopped, but when the first upgraded node boots up and connects to the database. Should make that clear.

File: site/docs/v1/tech/admin-guide/upgrade.adoc [hunk 3]
```diff
@@ -4,44 +4,373 @@ title: Upgrade FusionAuth
...
+
+This is a good option for single-instance development or QA systems, or for systems that can tolerate longer periods of downtime. For instance, if you know you have near-zero user activity at midnight, you can upgrade then. If you have scheduled downtime for other components of your system, that is a good time to upgrade FusionAuth in this manner.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/upgrade.adoc:148]: shoudl use oxford comma.

File: site/docs/v1/tech/admin-guide/upgrade.adoc [hunk 4]
```diff
@@ -4,44 +4,373 @@ title: Upgrade FusionAuth
...
+
+It's important to consider the timing implications of a rollback or any disaster recovery process. RTO, or Recovery Time Objective, is the targeted period time within which a service should be restored after an outage to avoid unacceptable consequences. Simply put, it is how long you can afford to be without the service or system before it severely impacts your business. It represents the goal for the time taken to recover from failure.
+
+RPO, or Recovery Point Objective, on the other hand, refers to the maximum tolerable amount of data loss measured in time. Determine RPO by looking at the time of the last backup you need to restore from and the restore time. 
+
+In the context of a FusionAuth upgrade rollback, the RPO would be the time since the last backup before the upgrade, and the RTO would be the time it takes to stand up the older version of the system and restore the database. 
+
+The RPO effectively measures the maximum tolerable data loss in the event of a rollback, and the RTO measures the time it takes to recover from the failure.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/upgrade.adoc:330]: This is in the admin guide :) .

File: site/docs/v1/tech/admin-guide/upgrade.adoc [hunk 5]
```diff
@@ -4,44 +4,373 @@ title: Upgrade FusionAuth
...
+- Script the changes using a client library.
+- Terraform.
+- Pulumi.
+- Manual configuration via the UI.
+
+You can read more in link:/docs/v1/tech/admin-guide/configuration-management[the Configuration Management guide].
+
+=== Themes and Customizations
+
+Review custom themes and how they will be impacted by any FusionAuth updates. If you have made any customizations, you may need to update them to work with the new version of FusionAuth. You can use the https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] project to perform diffs on the themes to identify any changes that need to be made.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/upgrade.adoc:131]: Good point, I think we don't need it.

File: site/docs/v1/tech/admin-guide/upgrade.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/upgrade.adoc:337]: Going to leave these terms in here in case someone is searching for them.

**Pull request merged**